workspaces/homepage/plugins/dynamic-home-page/src/components/EntitySection/EntitySection.tsx

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+import {
+  CodeSnippet,
+  WarningPanel,
+  Link as BackstageLink,
+} from '@backstage/core-components';
+
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import IconButton from '@mui/material/IconButton';
+import Skeleton from '@mui/material/Skeleton';
+import Typography from '@mui/material/Typography';
+import CloseIcon from '@mui/icons-material/Close';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import CircularProgress from '@mui/material/CircularProgress';
+import CardContent from '@mui/material/CardContent';
+import { useTheme, styled } from '@mui/material/styles';
+
+import EntityCard from './EntityCard';
+import { ViewMoreLink } from './ViewMoreLink';
+import HomePageEntityIllustration from '../../images/homepage-entities-1.svg';
+import { useEntities } from '../../hooks/useEntities';
+
+const StyledLink = styled(BackstageLink)(({ theme }) => ({
+  textDecoration: 'none',
+  padding: theme.spacing(1, 1.5),
+  fontSize: '16px',
+  display: 'inline-flex',
+  border: `1px solid ${theme.palette.primary.main}`,
+  borderRadius: 4,
+}));
+
+export const EntitySection = () => {
+  const theme = useTheme();
+  const [isRemoveFirstCard, setIsRemoveFirstCard] = React.useState(false);
+  const [showDiscoveryCard, setShowDiscoveryCard] = React.useState(true);
+  const [imgLoaded, setImgLoaded] = React.useState(false);
+
+  const isMediumScreen = useMediaQuery(theme.breakpoints.down(1280));
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down(1024));
+  const isExtraSmallScreen = useMediaQuery(theme.breakpoints.down(900));
+  let smallScreenWidth;
+  if (isExtraSmallScreen) {
+    smallScreenWidth = 266;
+  } else if (isSmallScreen) {
+    smallScreenWidth = 115;
+  } else {
+    smallScreenWidth = 180;
+  }
+  const imageWidth = isMediumScreen ? smallScreenWidth : 266;
```

Comment:
Is there no more responsive way to this?

I'm not saying css clamp needs to be used, but might a good fit, wdyt?

Maybe @logonoff can help with other ideas? :)